src/overlays/actors/ovl_En_Zov/z_en_zov.c

```diff
@@ -37,63 +40,593 @@ const ActorInit En_Zov_InitVars = {
+                func_80151938(globalCtx, globalCtx->msgCtx.unk11F04 + 1);
+                break;
+
+            case 0x1039:
+                globalCtx->nextEntranceIndex = globalCtx->setupExitList[ENZOV_GET_FE00(&this->actor)];
+                globalCtx->unk_1887F = 5;
+                globalCtx->sceneLoadFlag = 0x14;
+                gSaveContext.weekEventReg[78] |= 1;
+                this->actionFunc = func_80BD1D94;
+                globalCtx->msgCtx.unk11F10 = 0;
+                Audio_QueueSeqCmd(0x101400FF);
+                break;
+
+            default:
+                func_801477B4(globalCtx);
+                this->actionFunc = func_80BD1F1C;
+                break;
+        }
+    }
+}
+
+void func_80BD1F1C(EnZov* this, GlobalContext* globalCtx) {
+    func_80BD1764(this);
+
+    if (this->actor.home.rot.y != this->actor.shape.rot.y) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 2, 0x1000, 0x200);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        this->actionFunc = func_80BD1DB8;
+        func_80BD1D30(this, globalCtx);
+    } else if (func_80BD15A4(this, globalCtx)) {
+        func_800B8614(&this->actor, globalCtx, 120.0f);
+    }
+}
+
+s32 func_80BD1FC8(GlobalContext* globalCtx, EnZov* this) {
+    s32 ret;
+    s32 pad;
+
+    ret = func_8013A530(globalCtx, &this->actor, 4, &this->actor.focus.pos, &this->actor.shape.rot, 10.0f, 300.0f, -1);
+    ret |= func_8013A530(globalCtx, &this->actor, 5, &this->unk_308, &this->actor.shape.rot, 50.0f, 160.0f, 0x3000);
+    ret |= func_8013A530(globalCtx, &this->actor, 6, &this->unk_314, &this->actor.shape.rot, 50.0f, 160.0f, 0x3000);
+    return ret;
+}
+
+void EnZov_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnZov* this = THIS;
+
+    Actor_MoveWithGravity(&this->actor);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 10.0f, 10.0f, 10.0f, 4);
+
+    this->actionFunc(this, globalCtx);
+
+    if (!func_800EE29C(globalCtx, 0x1F8)) {
+        this->unk_320 &= ~0x10;
+    }
+
+    if ((this->unk_320 & 1) && func_80BD15A4(this, globalCtx)) {
+        func_800E9250(globalCtx, &this->actor, &this->unk_2F0, &this->unk_2F6, this->actor.focus.pos);
+    } else {
+        if ((this->unk_320 & 0x10) && (this->unk_322 == 0)) {
+            Math_SmoothStepToS(&this->unk_2F0.x, -0x1B58, 6, 0x1838, 0x64);
+        } else {
+            Math_SmoothStepToS(&this->unk_2F0.x, 0, 6, 0x1838, 0x64);
+        }
+        Math_SmoothStepToS(&this->unk_2F0.y, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->unk_2F6.x, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->unk_2F6.y, 0, 6, 0x1838, 0x64);
+    }
+
+    if (DECR(this->unk_2EE) == 0) {
+        this->unk_2EE = Rand_S16Offset(0x3C, 0x3C);
```

Comment:
```suggestion
        this->unk_2EE = Rand_S16Offset(60, 60);
```
This looks like a timer